src/audio/src/src_generic.c

```diff
@@ -60,21 +60,15 @@ static inline void fir_filter_generic(int32_t *rp, const void *cp, int32_t *wp0,
 		 * Qx.31.
 		 */
 		for (i = 0; i < n1; i++) {
-			y0 += (int64_t)(*coef) * (*data);
-			data++;
-			y1 += (int64_t)(*coef) * (*data);
-			data++;
-			coef++;
+			y0 += (int64_t)(*coef) * (*(data++));
+			y1 += (int64_t)(*(coef++)) * (*(data++));
 		}
 		if (data == fir_end)
 			data = fir_start;
 
 		for (i = 0; i < n2; i++) {
-			y0 += (int64_t)(*coef) * (*data);
-			data++;
-			y1 += (int64_t)(*coef) * (*data);
-			data++;
-			coef++;
+			y0 += (int64_t)(*coef) * (*(data++));
+			y1 += (int64_t)(*(coef++)) * (*(data++));
```

Comment:
I would suggest remove (), due to () may bring understanding trouble to developer, * or () which have higher priority?
y1 += (int64_t)(*coef++ * *data++);

Reply:
we seem to be in a rather subjective area here. My personal order of preference:
1. `for (i=0; i<5; i++, ptr++) y=*ptr;`
2. `for (i=0; i<5; i++) {y=*ptr; ptr++;}`
3. `for (i=0; i<5; i++) y=*(ptr++);`
4. `for (i=0; i<5; i++) y=*ptr++;`

but that's just me. I can live with either of them, so please choose whatever you, guys, who work with this code most frequently, feel most comfortable with. I also don't think this choice would make any performance difference. Possibly only in loops like the one above, where `data` is incremented twice. *Maybe* doing
```
for (i=0; i<5; i++) {
	y0 += data[0];
	y1 += data[1];
	data += 2;
}
```
would generate 1 or 2 asm instructions fewer, or maybe it won't

Reply:
Fwiw, the latter from @lyakh comment is probably easier for compilers to track/vectorize as the offsets are constant wrt loop iteration.